### PR TITLE
Update Bond Comm docs

### PR DESCRIPTION
**Pull request by chadwalters**
Include initial C++ release and updates to C# implementation. Also,
update minimum version of Boost to 1.58.

### CODE REVIEW

File: doc/src/bond_cs.md [hunk 1]
```diff
@@ -1227,12 +1227,9 @@ Today the framework supports two messaging patterns:
...
 idiomatic Task and async/await facilities of .NET.
 
-The initial release of Bond Comm is designed to run on Windows running
-.NET&nbsp;4.5 or .NET&nbsp;Core. Since .NET&nbsp;Core is still under development, we
-will update as needed. Mono support is forthcoming, as are Linux and
-Mac&nbsp;OS&nbsp;X.
-
-Only Windows 10 and Windows Server 2012 R2 have been tested to date.
+Bond Comm is designed to run on NET&nbsp;4.5 or .NET&nbsp;Core. Mono
+support is forthcoming, as are Linux and Mac&nbsp;OS&nbsp;X -- only Windows 10 and Windows
```
Comment (chwarr) [inline doc/src/bond_cs.md]: "macOS" here and elsewhere?

**Comment by chadwalters** [comment]
CHANGELOG updated to reflect bump to Boost 1.58 as minimum version.

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,6 +17,7 @@ different versioning scheme, following the Haskell community's
 
 * Fixed compatibility with RapidJSON v1.1.0.
   [Issue #271](https://github.com/Microsoft/bond/issues/271)
+* Set Boost 1.58 as minimum supported version
```
Comment (chwarr) [inline CHANGELOG.md]: "The minimul support version of Boost is now 1.58"?

Comment (chadwalters) [inline CHANGELOG.md]: OK, adjusted.

**Pull request merged**